src/gui/layout/qgslayoutview.cpp

```diff
@@ -1120,7 +1178,9 @@ void QgsLayoutView::keyReleaseEvent( QKeyEvent *event )
 void QgsLayoutView::resizeEvent( QResizeEvent *event )
 {
   QGraphicsView::resizeEvent( event );
+  disconnect( this, &QgsLayoutView::zoomLevelChanged, this, &QgsLayoutView::extentChanged );
   emit zoomLevelChanged();
+  connect( this, &QgsLayoutView::zoomLevelChanged, this, &QgsLayoutView::extentChanged );
```

Comment:
This looks wrong -- won't a resize generally cause a different extent? (eg when resizing to a different aspect ratio).

Reply:
Yes, but actually, the term `extentChanged` is a bit misleading. What matters is only the center coordinates and the transform, and those won't change if the view is resized.

The problem I had was calling zoomNext or zoomLast can make the scrollbar appear or disappear, which triggers a resizeEvent, creating zoom history wipe or loops. 

Also, not sure what is the `emit zoomLevelChanged();` purpose here ; should the cache really be invalidated when the view is resized? If not, we can just get rid off those three lines.

Reply:
> The problem I had was calling zoomNext or zoomLast can make the scrollbar appear or disappear, which triggers a resizeEvent, creating zoom history wipe or loops.

I'd handle that by adding a timer in the code which adds an entry to the zoom history. If the last entry was added < say 1 second ago, then we ignore the new one (or even better, replace the previous one!). That would also help with the case of say zooming in using a mouse wheel + ctrl key, which will fire many zoom level changes in a rapid period of time.

> Also, not sure what is the emit zoomLevelChanged(); purpose here ; should the cache really be invalidated when the view is resized? If not, we can just get rid off those three lines.

Yes, it should. It triggers re-updating layout items at the new zoom level, because otherwise we'd draw them in a pixelated resized version of their previous render.

Reply:
I must be missing something, but I don't see how the resize impacts the zoom level. 
![qgis-bin_SV5Kqe3JAY](https://github.com/user-attachments/assets/9c180c72-e383-4ca0-adb8-4fb75fe5ae7f)

Reply:
You're right, sorry! We don't need the zoomElevelChanged signal there.